### ISSUE TITLE
[BUG] Redis Backed Chat Memory Error - "Cannot read private member from an object whose class did not declare it"

### ISSUE THREAD

**Issue by ezynda3** [FlowiseAI/Flowise#614]
**Describe the bug**
When using Redis Backed Memory Store in a Conversational QA Retriever Chain I get the following error:
```sh
TypeError: Cannot read private member from an object whose class did not declare it
    at __classPrivateFieldGet (/usr/local/lib/node_modules/flowise/node_modules/@redis/client/dist/lib/client/index.js:4:94)
    at get isReady [as isReady] (/usr/local/lib/node_modules/flowise/node_modules/@redis/client/dist/lib/client/index.js:86:16)
    at RedisChatMessageHistory.ensureReadiness (/usr/local/lib/node_modules/flowise/node_modules/langchain/dist/stores/message/redis.cjs:48:26)
    at RedisChatMessageHistory.getMessages (/usr/local/lib/node_modules/flowise/node_modules/langchain/dist/stores/message/redis.cjs:54:20)
    at BufferMemory.loadMemoryVariables (/usr/local/lib/node_modules/flowise/node_modules/langchain/dist/memory/buffer_memory.cjs:40:49)
    at ConversationalRetrievalQAChain.call (/usr/local/lib/node_modules/flowise/node_modules/langchain/dist/chains/base.cjs:80:49)
    at ConversationalRetrievalQAChain_Chains.run (/usr/local/lib/node_modules/flowise/node_modules/flowise-components/dist/nodes/chains/ConversationalRetrievalQAChain/ConversationalRetrievalQAChain.js:161:37)
    at App.processPrediction (/usr/local/lib/node_modules/flowise/dist/index.js:723:42)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /usr/local/lib/node_modules/flowise/dist/index.js:427:13
```

**To Reproduce**

1. Go to Marketplace and choose "Conversational Retrieval QA Chain".
2. Click on 'Use Template'
3. Add a Redis Backed Chat Memory and plug it into the memory slot of the "Conversational Retrieval QA Chain" card.
4. Configure the redis card to point to a valid Redis instance.
5. Save
6. Run chat as normal and you will see an error.

**Expected behavior**
Chat should function normally without error.

**Screenshots**
![image](https://github.com/FlowiseAI/Flowise/assets/5308871/4981e3af-4cdb-44b9-bdba-a955074d5cb7)

**Setup**
-   Installation docker on Portainer instance
-   Flowise Version 1.2.16
-   OS: Linux
-   Browser Brave

**Comment by HenryHengZJ** [comment]
does it work if you use the conversation chain:
![image](https://github.com/FlowiseAI/Flowise/assets/26460777/c27e51b1-7b16-4c38-9aa3-1d7d6900eedb)

**Comment by ezynda3** [comment]
Yes. Works just fine there.

**Comment by magnum6actual** [comment]
I'm hitting this same error with redis with an OpenAI function agent.  The odd thing - it works fine if I query from the Flowise UI.  If it send a query via API, it throws the "Cannot read private member from an object whose class did not declare it".

**Comment by giquieu** [comment]
Hello, I have the same return.
Using the Flowise UI chat, there is feedback. If you use the API the error happens.

**Comment by aleksanterix** [comment]
Hello, I'm having the same problem with the Conversational Retrieval QA Chain.

**Comment by vijaykammili** [comment]
I have the same issue with the Conversational Retrieval QA Chain as well.

**Comment by vijaykammili** [comment]
@HenryHengZJ or  @chungyau97 - Can you help fix this issue please?

**Comment by aleksanterix** [comment]
I personally used the "Upstash Redis backed chat memory" to resolve this issue.

Please tag me if the issue with Redis backed chat memory gets resolved.

--
Aleksanteri Halonen
Junior DevOps Engineer
+358 50 463 0572
Avanio

**Comment by HenryHengZJ** [comment]
This PR will be fixing the issue - https://github.com/FlowiseAI/Flowise/pull/1118

**Comment by vinodkiran** [comment]
Fixed in [Release 1.3.9](https://github.com/FlowiseAI/Flowise/releases/tag/flowise%401.3.9)
